wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -154,24 +155,40 @@
   });
 
   private final Schema schema;
+  private final ImmutableMap<ProtoType, TypeName> nameToWireJavaName;
```

Comment:
. . . instead you could probably change the code to prefer the configured name if it exists